python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -366,19 +367,24 @@ class RandomForestClassifier(Base):
         self._reset_forest_data()
 
     def _reset_forest_data(self):
-        # Only if model is fitted before
-        # Clears the data of the forest to prepare for next fit
-        if self.n_cols:
+        """Free memory allocated by this instance and clear instance vars."""
```

Comment:
Can `_reset_forest_data` be pulled into a shared cython file?

Reply:
We are hoping for a big refactor to break out some of the common code (@Salonijain27 has started on this). Many of the methods like this one are completely identical _except_ for the datatypes of RandomForestMetaData which differ between classifier and regressor. She is currently experimenting with options to wipe out this complexity, which may end up being partial (e.g. maybe the delete_rf_metadata stays per-class but all the other code is factored out?). Based on that, I'd prefer to keep the common code breakouts for that refactor.

(But believe me, nobody is more supportive of factoring these out than I am. Drives me crazy to basically copy paste code and change one datatype, and I mess it up _all_ the time.)